3D Chess/Assets/Scripts/ResetButton.cs

```diff
@@ -0,0 +1,11 @@
+using UnityEngine;
+
+public class ResetButton : MonoBehaviour
+{
+    [SerializeField] private GameState gameState = default;
```

Comment:
Do we need the `= default` for all of these `[SerializeField]` fields? I would assume that by default they're assigned to the default value for the type.

Reply:
Unfortunately (or fortunately?) we do. Visual Studio will show warning for fields that have never been assigned to because it does not understand what's going on with Unity correctly. By adding `default` and therefore explicitly assigning a value, we avoid this. It's common practice in Unity development and many, many forum posts can be found with this recommendation.

Another good example is this (also look at the reference count):

![image](https://user-images.githubusercontent.com/1942012/128187107-de7356a6-71ef-44a5-ac8e-486f0dc8dad3.png)

Reply:
Cool, if it's a common practice, let's stick to it.